network/openssl_privatekey.py

```diff
@@ -0,0 +1,225 @@
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+
+try:
+    from OpenSSL import crypto
+except ImportError:
+    pyopenssl_found = False
+else:
+    pyopenssl_found = True
+
+
+import os
+
+DOCUMENTATION = '''
+---
+module: openssl_privatekey
+author: "Yanis Guenane (@Spredzy)"
+version_added: "2.2"
+short_description: Generate OpenSSL private keys
+description:
+    - "This module allows one to (re)generates OepnSSL private keys. It uses
+       the pyOpenSSL python library to interact with openssl. One can generate
+       either RSA or DSA private keys. Keys are generated in PEM format."
+requirements:
+    - "python-pyOpenSSL"
+options:
+    state:
+        required: false
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the private key should exist or not, , taking action if the state is different from what is stated.
+    name:
+        required: true
+        description:
+            - Name of the generated OpenSSL private key
+    size:
+        required: false
+        default: 4096
+        description:
+            - Size of the SSL key go generate
+    type:
+        required: false
+        default: "RSA"
+        choices: [ RSA, DSA ]
+        description:
+            - The algorithm to use to generate the SSL private key
+    force:
+        required: false
+        default: False
+        choices: [ True, False ]
+        description:
+            - Should the key be forced force by this ansible module
+    path:
+        required: true
+        description:
+            - Name of the folder in which the generated SSL private key will be written
+'''
+
+EXAMPLES = '''
+# Generate an OpenSSL private key with the default values (4096 bits, RSA)
+- openssl_privatekey: name=ansible.com.pem path=/etc/ssl/private
+
+# Generate an OpenSSL private key with a different size (2048 bits)
+- openssl_privatekey: name=ansible.com.pem path=/etc/ssl/private size=2048
+
+# Force regenerate an OpenSSL private key if it already exists
+- openssl_privatekey: name=ansible.com.pem path=/etc/ssl/private force=True
+
+# Generate an OpenSSL private key with a different algorithm (DSA)
+- openssl_privatekey: name=ansible.com.pem path=/etc/ssl/private type=DSA
+'''
+
+RETURN = '''
+size:
+    description: Size of the SSL private key
+    returned:
+        - changed
+        - success
+    type: integer
+    sample: 4096
+type:
+    description: Algorithm used to generate the SSL private key
+    returned:
+        - changed
+        - success
+    type: string
+    sample: RSA
+privatekey:
```

Comment:
This is a bit strange to me; it feels as if "privatekey" ought to contain the key itself.  Maybe rename to "filename"?